### ISSUE TITLE
Depreciation with half-year convention

### ISSUE THREAD

**Issue by gmarcon** [snipe/snipe-it#1237]
Hi,

  could you please consider a feature request for half-year convention in depreciation?
(https://en.wikipedia.org/wiki/Half-year_convention)

Basically, if an asset depreciates over a 5 years/60 months period, the first year the asset is depreciated of a 10% of its value, year 2 to 5 20% each year, and year 6 10% of its value.

This depreciation method is common also in Europe (for instance it is the only one allowed in Italy and Austria).

Thank you and regards,
Giulio

**Comment by gmarcon** [comment]
Commenting to avoid closure by stalebot: could you please consider adding this feature? Right now to have the correct assets evaluation we have to export the data and calculate the asset value in an Excel file. It would be very practical to have the feature directly in Snipe-IT.

**Comment by uberbrady** [comment]
Well the absolute guaranteed-fastest way to get this feature would be for you to write it, yourself - and we'd definitely be happy to accept the PR if you were to do that. Even if it needed some cleanup. We'll be nice, I promise! Especially if you say something like, "It's my first Pull Request, please be gentle!" :)

If you're not a programmer-type (and of course, plenty of people aren't!) you can also use the 'Bountysource' links to apply a bounty to the issue. That way, when a programmer solves it in a pull request, they get a payout. Effectively, you'd be "paying" for the feature.

**Comment by uberbrady** [comment]
But anyways, I had some questions - 

1) If you use this depreciation method, do you use it on *every* thing you buy? Because in Snipe-IT, depreciation methods are associated with asset models, not with individual assets. So if you bought one "MacBook Pro 15-inch" near July 1, and another near January 15 - would you want that method to be applied to both, or only to the mid-year one?
2) Do you do the depreciation only at the mid-year mark? E.g. - 

$1,000 computer (for simplicity) - 5 year, half-year convention - purchased July 1 2018 . At 5 years, you are deprecating 20% per year, or 10% in a half-year 

| Dates | deprecation value | remaining value | notes
|-------|-------------------|----------------|-----
| July-Dec 2018 | 0 | $1000 | initial value
|Jan 2019-Jun 2019 | $100 | $900 | Year ZERO second half-year deprecation fired
| July 2019-Dec 2019 | $200 | $800 | Year ONE first half-year depreciation fired
| Jan 2020-Jun 2020 | $300 | $700 | Year ONE second half-year fired
| Jul 2020-Dec 2020 | $400 | $600 | Year TWO first half-year depreciation fired
| Jan 2021-Jun2021 | $500 | $500 | Year TWO second half-year deprecation fired
| Jul 2021-Dec2021 | $600 | $400 | 
| Jan 2022-Jun2022 | $700 | $300 |
| Jul 2022-Dec 2022 | $800 | $200|
| Jan 2023-Jun 2023| $900 | $100|
| Jul 31 2023 | $1000 | $0| Only after June 30 would the final deprecation be taken?

So, in other words, does the deprecation only 'show up' if you look after December 31? I assume you're getting your taxes ready much *before* December 31, so that seems like it would be weird.

I'm not sure. Anyways, I thought we should at least talk through how the thing should work.

**Comment by gmarcon** [comment]
Thank you for your feedback.

The half-year convention for depreciation has actually different implementations for each country depending on the fiscal policy, for instance in Italy it is always applied, in Austria it is applied only if the purchase happened in the second half of the year.

The depreciation of a 1.000€ asset with 5 years in Austria, if bought in the second half of the year would be:
- Acquisition July 1st 2018, value 1.000€
- Value at Dec 2018: 900€ (depreciation 100€)
- Value at Dec 2019: 700€ (depreciation 300€)
- Value at Dec 2020: 500€ (depreciation 500€)
- Value at Dec 2021: 300€ (depreciation 700€)
- Value at Dec 2022: 100€ (depreciation 900€)
- Value at Dec 2023: 0€ (depreciation 1.000€)

The depreciation of a 1.000€ asset with 5 years in Austria, if bought in the first half of the year would be:
- Acquisition February 1st 2018, value 1.000€
- Value at Dec 2018: 800€ (depreciation 200€)
- Value at Dec 2019: 600€ (depreciation 400€)
- Value at Dec 2020: 400€ (depreciation 600€)
- Value at Dec 2021: 200€ (depreciation 800€)
- Value at Dec 2022: 0€ (depreciation 1.000€)

In Italy, independently from the date when the asset has been bought, the half-year convention applies:
- Acquisition February 1st 2018, value 1.000€
- Value at Dec 2018: 900€ (depreciation 100€)
- Value at Dec 2019: 700€ (depreciation 300€)
- Value at Dec 2020: 500€ (depreciation 500€)
- Value at Dec 2021: 300€ (depreciation 700€)
- Value at Dec 2022: 100€ (depreciation 900€)
- Value at Dec 2023: 0€ (depreciation 1.000€)

The monthly depreciation follows the yearly convention divided by 12, but is not relevant for fiscal purposes as fiscal depreciation is calculated yearly, it could be useful only for trial balance purposes.

Regards,
Giulio

**Comment by gmarcon** [comment]
I put a $100 bounty on this issue, anyone interested in developing it?

**Comment by aaronvasic** [comment]
Hey, I'm taking a look at it. This looks interesting to me. I've built financial software in the past. I'll update when I've got more worked out.

**Comment by VELIKII-DIVAN** [comment]
gmarcon, I'd like to do it. Could you explain in more detail? 
1 Would you like to change the depreciation report page ([https://demo.snipeitapp.com/reports/depreciation](https://demo.snipeitapp.com/reports/depreciation))? 
Also, I have to add some settings to switch depreciation method. But, I don't use this application and don't know where is the good place for the switch setting. Could you specify?

**Comment by gmarcon** [comment]
There was apparently a merge to the develop branch, I am unsure if this already made it into a release/has been tested. I could test it myself if there is an available continuous integration environment where the develop branch is deployed.

**Comment by VELIKII-DIVAN** [comment]
> There was apparently a merge to the develop branch, I am unsure if this already made it into a release/has been tested. I could test it myself if there is an available continuous integration environment where the develop branch is deployed.

There is the develop branch [develop.snipeitapp.com](develop.snipeitapp.com)
Depreciation settings [develop.snipeitapp.com/admin/settings](develop.snipeitapp.com/admin/settings)
